core/src/main/kotlin/com/alecstrong/sql/psi/core/psi/mixins/CompoundSelectStmtMixin.kt

```diff
@@ -41,17 +43,21 @@ internal abstract class CompoundSelectStmtMixin(
 
   override fun tablesAvailable(child: PsiElement): Collection<LazyQuery> {
     val tablesAvailable = super.tablesAvailable(child)
-    val parent = parent
-    if (parent is SqlWithClause) {
-      if (parent.node.findChildByType(
-              SqlTypes.RECURSIVE) != null &&
-          child != selectStmtList.first()) {
-        return tablesAvailable + parent.tablesExposed()
-      }
-      val myIndex = parent.compoundSelectStmtList.indexOf(this)
-      return tablesAvailable + parent.tablesExposed().filterIndexed { index, _ -> index != myIndex }
+    val withClauseAuxiliaryStmts = parent as? SqlWithClauseAuxiliaryStmt ?: return tablesAvailable
+    val withClause = withClauseAuxiliaryStmts.parent as SqlWithClause
+    if (withClause.node.findChildByType(SqlTypes.RECURSIVE) != null && child != selectStmtList.first()) {
+      return tablesAvailable + withClause.tablesExposed()
     }
-    return tablesAvailable
+    val myIndex = withClause.withClauseAuxiliaryStmtList
+      .mapNotNull { withClauseAuxiliaryStmt ->
+        PsiTreeUtil.findChildOfAnyType(
```

Comment:
could this be `withClauseAuxiliaryStmt.children.filter { it is QueryElement }`? (to avoid dep on Postgres)

Reply:
Fixed in https://github.com/AlecStrong/sql-psi/pull/170/commits/0647c327e733e690fba8344a4bc4c244e24f7dc9.

I can't directly filter the children as the `ReturningClause` is a `QueryElement` but is a "grandchild" of `withClauseAuxiliaryStmt`.